internal/beater/config/agentconfig.go

```diff
@@ -121,11 +118,11 @@ func (s *FleetAgentConfig) setup() error {
 		s.Config = make(map[string]string)
 	}
 	if s.Etag == "" {
-		m, err := json.Marshal(s)
+		etag, err := generateEtag(s)
 		if err != nil {
-			return fmt.Errorf("error generating etag for %s: %v", s.Service, err)
+			return err
 		}
-		s.Etag = fmt.Sprintf("%x", md5.Sum(m))
+		s.Etag = etag
```

Comment:
What do you think of not set Etag at all in FIPS scenario?

As the function already takes a `*FleetAgentConfig`, we could set `s.Etag` only in the non-FIPS implementation.

Something like
```
if s.Etag == "" {
err := generateEtag(s)
}
```

and then setting `s.Etag = etag` in the implementation.